### PR TITLE
docs: Move cve_scan and generation of external dependencies rst to bazel

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs: Move generation of external dependencies rst to bazel
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: tools/dependency/utils.py
Comment (phlax) [inline tools/dependency/utils.py:18]: `"api/"`

File: tools/extensions/BUILD
Comment (phlax) [inline tools/extensions/BUILD:24]: ?

File: tools/dependency/BUILD
Comment (phlax) [inline tools/dependency/BUILD:13]: ?

File: tools/dependency/utils.py
Comment (phlax) [reply tools/dependency/utils.py:18]: actuall this is the only place it is used so the old default doesnt matter

```
$ git grep dependency | grep utils
tools/dependency/generate_external_dep_rst.py:from tools.dependency import utils as dep_utils
```

File: docs/build.sh
Comment (phlax) [inline docs/build.sh:55]: leave this for now

File: tools/dependency/cve_scan.py
Comment (phlax) [inline tools/dependency/cve_scan.py:16]: add comment here

File: tools/dependency/utils.py
Comment (phlax) [reply tools/dependency/utils.py:18]: actually not - that was incorrect - the cve_scan module imports it...

**Comment by phlax** [comment]
generated rst is here:

https://storage.googleapis.com/envoy-pr/bbaf31b/docs/intro/arch_overview/security/external_deps.html

**Comment by ggreenway** [comment]
I'm not the right reviewer for this; I don't know bazel well enough. Maybe @lizan ?

**Comment by phlax** [comment]
cc @htuch @moderation

**Comment by phlax** [comment]
thanks @ggreenway

**Comment by moderation** [comment]
This LGTM but definitely needs @htuch with his combination of Bazel and Python skills to review

File: tools/dependency/BUILD
Comment (phlax) [inline tools/dependency/BUILD:6]: not sure if this should be an `envoy_package()`

Comment (htuch) [reply tools/dependency/BUILD:6]: This is fine, it's just setting visibility (you can skip it on the targets).

File: tools/dependency/cve_scan.py
Comment (htuch) [inline tools/dependency/cve_scan.py:10]: Seems unrelated and that it should have been broken before if not there.

File: tools/dependency/generate_external_dep_rst.py
Comment (htuch) [inline tools/dependency/generate_external_dep_rst.py:14]: Oh, I see. How complex is it to make Bazelify cve_scan?

**Review by htuch — COMMENTED** [review]
Looks good, but would be idea if we didn't need any special casing on the border of Bazel and non-Bazel Python use..

File: tools/dependency/generate_external_dep_rst.py
Comment (phlax) [reply tools/dependency/generate_external_dep_rst.py:14]: not too complex i think

would you prefer to update that first ?

File: tools/dependency/BUILD
Comment (phlax) [reply tools/dependency/BUILD:6]: targets updated

File: tools/dependency/cve_scan.py
Comment (phlax) [reply tools/dependency/cve_scan.py]: removed

File: tools/dependency/generate_external_dep_rst.py
Comment (phlax) [reply tools/dependency/generate_external_dep_rst.py:14]: pr added for `cve_scan` here #15342

**Comment by phlax** [comment]
one hesitation i have with this PR is that i think it would be better for the tools in `/tools/extension` and `/tools/dependency` to all just generate eg json - and then have that all put together into rst in a doc tool

i think we can do that once we have finished moving the docs generation to bazel tho

File: ci/run_envoy_docker.sh
Comment (phlax) [inline ci/run_envoy_docker.sh:49]: these wont be necessary once #15358 lands

File: tools/dependency/generate_external_dep_rst.py
Comment (phlax) [reply tools/dependency/generate_external_dep_rst.py:14]: and merged with this PR to simplify...

**Comment by htuch** [comment]
@phlax JSON (or proto) for intermediate artifacts makes sense to me.

**Review by htuch — COMMENTED** [review]
LGTM for draft, please ping when ready for full review.
/wait

**Comment by phlax** [comment]
@htuch ping - should be ready for final review (nothing changed since last review)

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by htuch** [comment]
@phlax CI seems stuck, can you merge main and push?

**Comment by moderation** [comment]
@phlax I run a pretty heavily modified version of `tools/dependency/release_dates.py` multiple times a day as part of deps shepharding. I think this PR introduced a breakage.
```
python3.9 tools/dependency/release_dates.py api/bazel/repository_locations.bzl | rg -vi github.re
Traceback (most recent call last):
  File "/home/moderation/Library/envoyproxy/envoy/tools/dependency/release_dates.py", line 156, in <module>
    spec_loader = utils.repository_locations_utils.load_repository_locations_spec
AttributeError: module 'utils' has no attribute 'repository_locations_utils'
```
Should I be running this now via `bazel`?

**Comment by moderation** [comment]
@mattklein123 was seeing CI failures and we discussed in Slack. The PR has been reverted to unblock CI - https://github.com/envoyproxy/envoy/pull/15421

**Pull request merged**
